src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -257,21 +258,17 @@ class ReadFromObjectStorageStep : public SourceStepWithFilter
     void applyFilters(ActionDAGNodes added_filter_nodes) override
     {
         SourceStepWithFilter::applyFilters(std::move(added_filter_nodes));
-        const ActionsDAG::Node * predicate = nullptr;
         if (filter_actions_dag.has_value())
         {
-            predicate = filter_actions_dag->getOutputs().at(0);
             if (getContext()->getSettingsRef()[Setting::use_iceberg_partition_pruning])
-            {
                 configuration->implementPartitionPruning(*filter_actions_dag);
```

Comment:
Let's get rid of this interface in the next PR?

Reply:
Will do.